src/Services/PatientService.php

```diff
@@ -173,6 +175,77 @@ public function databaseInsert($data)
         }
     }
 
+    /**
+     * Grants portal access to patients in case they do
+     * not have the permission.
+     *
+     * @param $puuidString
+     * @return void
+     */
+    public function allowPatientPortal($puuidString)
+    {
+        $sql = "UPDATE `patient_data` SET";
+        $sql .= " `allow_patient_portal`='YES'";
+        $sql .= " WHERE `uuid` = ?";
+        $puuidBinary = UuidRegistry::uuidToBytes($puuidString);
+        sqlQuery($sql, array($puuidBinary));
+    }
+
+    /**
+     * Verifies if the patient has access to the portal.
+     *
+     * @param $patient_data
+     * @return boolean
+     */
+    public function hasPortalPermission($patient_data)
+    {
+        $patient_portal_permission = strtolower($patient_data["allow_patient_portal"]);
+        $has_portal_permission = "yes";
+
+        return !(
+            empty($patient_data["allow_patient_portal"]) ||
+            $patient_portal_permission != $has_portal_permission
+        );
+    }
+
+    /**
+     * This function is responsible for adding a password for
+     * patients to access the portal. If no password is specified,
+     * the default password is the document number.
+     *
+     * @param $patient_data
+     * @return ProcessingResult
+     */
+    public function setPatientAccess($puuidString, $data)
+    {
+        $data["uuid"] = $puuidString;
+        $portal_permission = 1;
+        $patientAccessOnSiteService = new PatientAccessOnsiteService(); 
+        $processingResult = $this->patientValidator->validate($data, PatientValidator::DATABASE_UPDATE_CONTEXT);
+        if (!$processingResult->isValid()) {
+            return $processingResult;
+        }
+        
+        $patient = $this->getOne($puuidString);
+        if($patient->hasData()) {
```

Comment:
If this "if" setence doesnt work, what is the default value to return for the method?

Reply:
If the condition does not enter the 'if' statement, a 404 will be returned since the only scenario in which the condition cannot be accessed is if the patient does not exist